### PR TITLE
Ray 1.4 support

### PR DESCRIPTION
**Pull request by grechaw**
This PR updates the project so that it runs with ray 1.4.0.
I've done just basic testing by running the existing DAGs -- they are successful.
Although I've not managed to run the modin dag as my network doesn't love that example from my home.

### CODE REVIEW

File: ray_provider/hooks/ray_client.py [hunk 1]
```diff
@@ -28,7 +28,7 @@ def get_conn(self) -> str:
         if self.ray_conn_id:
             conn = self.get_connection(self.ray_conn_id)
 
-        if conn.host and "://" in conn.host and self.schema:
+        if conn.host and "://" in conn.host and conn.schema:
```
Comment (grechaw) [inline ray_provider/hooks/ray_client.py:31]: I think this was just a bug --- 'self.schema' threw an exception when  I triggered this IF

File: ray_provider/hooks/ray_client.py [hunk 2]
```diff
@@ -45,13 +45,17 @@ def connect(self):
...
+        log.info(f"Connection base_url is {self.base_url}")
+        log.info(f"Ray version is {ray.__version__}")
         # currently there isn't much useful info
         # returned from ray.util.connect(),
         # but if there could be, here would be where to use it,
         # so we should work to understand that as well.
         if not ray.util.client.ray.is_connected():
-            ray.util.connect(self.base_url)
+            if ray.__version__ == "1.4.0":
+                ray.client(self.base_url).namespace("airflow").connect()  # let's see if this changes things
```
Comment (richardliaw) [inline ray_provider/hooks/ray_client.py:56]: can we do version >= 1.4.0?

Comment (grechaw) [reply ray_provider/hooks/ray_client.py:56]: yes of course -- does one do just string comparison for versions or is there some other python idiom?

Comment (grechaw) [reply ray_provider/hooks/ray_client.py]: I've updated this

**Comment by grechaw** [comment]
Richard, I updated the version string as you suggested.  I suppose it will work until ray 1.10.0

In my further work to get this running on anyscale I've hit an architectural issue that we might want to think over before proceeding with more integration work -- basically the various tasks when run in parallel all start up new jobs (on the same cluster) and cannot get a lock on the cluster.

File: ray_provider/hooks/ray_client.py (hunk 2 above)
Comment (richardliaw) [reply ray_provider/hooks/ray_client.py]: hmm usually we will use LooseVersion object:  https://stackoverflow.com/questions/11887762/how-do-i-compare-version-numbers-in-python

**Comment by marrocksd** [comment]
@richardliaw sorry to bother but could you review that small change please, I have been waiting. Thank you alot

**Review by richardliaw — APPROVED** [review]

**Comment by richardliaw** [comment]
@marrocksd done!

**Pull request merged**